routes/order.router.js

```diff
@@ -0,0 +1,56 @@
+const express = require("express");
+const router = express.Router();
+
+const Order = require("../models/order.model");
+const Cart = require("../models/cart.model");
+
+router.get("/", async (req, res) => {
+  try {
+    const orders = await Order.find({ uid: req.uid }).populate({
+      path: 'products.product'
+    });
+    res.json({
+      success: true,
+      orders: orders
+    });
+  } catch (e) {
+    res.status(500).json({
+      error: {
+        message: "Mongoose error: " + e.message
+      }
+    })
+  }
+});
+
+router.post("/", async (req, res) => {
+  try {
+    const cart = await Cart.findOne({ uid: req.uid });
+    const populatedCart = await cart.populate({
+      path: 'cartLines.product'
+    }).execPopulate();
+    const totalPrice = getTotalPrice(populatedCart);
+    const order = await Order.create({ uid: req.uid, products: cart.cartLines, totalPrice: totalPrice });
+    cart.cartLines = [];
+    const updatedCart = await cart.save();
```

Comment:
As you are not sending this updatedCart variable in res.json, so you can directly do await cart.save

Reply:
I've made the change as suggested